project/views.py

```diff
@@ -0,0 +1,19 @@
+from django.shortcuts import render, redirect
+from django.utils import timezone
+from django.shortcuts import render
+from django.http import HttpResponseRedirect
+from .forms import habitForm
+from .models import habitEntry
+
+# Create your views here.
+def homepage(request):
+    if request.user.is_authenticated:
+        return render(request, "project/template/tracker.html")
+
+def habitForm(request):
```

Comment:
what do you want this view to do? this is mapped to this url `project/`, which looks like it does not have a clear role. 🤔 

here are the urls you need -- and you will need views to go with them:
- `/habits` # a list of all a users's habits
- `/habit/<int:pk>` # one habit (and you should show its associated records here as well)
-  `habit/<int:pk>/<int:year>/<int:month>/<int:day>` # a record for a given day for a certain habit (in this view you should be able to either retrieve the record or create a new one -- that is tricky; you can use `get_or_create` https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.get_or_create)

If you have that core set working then you can work on forms to add a new habit and edit a habit, and delete a habit.